docs/_source/docs/resolvers.rst

```diff
@@ -79,6 +79,37 @@ A resolver to execute any shell command.
 
 Refer to `sceptre-resolver-cmd <https://github.com/Sceptre/sceptre-resolver-cmd/>`_ for documentation.
 
+stack_attr
+~~~~~~~~~~
+
+This resolver lets you reference the values of other fields on the same Stack Config or those
+inherited from StackGroups in which the Stack Config exists. It is useful for avoiding unnecessary
+configuration duplication as well as referencing attributes inherited from the StackGroup Config.
+
+For example, if you need to reference the ``template_bucket_name`` set in the StackGroup Config,
+you can do that with ``!stack_attr``:
+
+.. code-block:: yaml
+
+   template:
+       type: sam
+       path: path/from/my/cwd/template.yaml
+       # template_bucket_name could be set by a resolver in the StackGroup.
+       artifact_bucket_name: !stack_attr template_bucket_name
+
+The argument to this resolver is the full attribute "path" from the Stack Config. You can access
+nested values in dicts and lists using "." to separate key/index segments. For example:
+
+.. code-block:: yaml
+
+   sceptre_user_data:
+       key:
+           - "some random value"
+           - "the value we want to select"
+
+   parameters:
+       my_parameter: !stack_attr sceptre_user_data.key.1
```

Comment:
does ordering matter?  can you put paramters above sceptre_user_data?

Would it be possible to use more concrete examples for `key` and the values for it?   Also could we add a comment for `my_parameter` telling users what  `!stack_attr sceptre_user_data.key.1` will pass to my_parameter?

Reply:
> does ordering matter? can you put parameters above sceptre_user_data?

Sure? I'm not sure why it matters. Remember render time is prior to resolve time, so the order they appear in on the template doesn't really matter. In this particular case, I wanted to introduce the data we're selecting first before selecting it. 

> Would it be possible to use more concrete examples for key and the values for it? Also could we add a comment for my_parameter telling users what !stack_attr sceptre_user_data.key.1 will pass to my_parameter?

Yeah, I could definitely do that.